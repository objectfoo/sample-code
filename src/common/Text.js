import React from 'react';
import injectStyle from 'react-jss';
import classnames from 'classnames';

export const Text = ({
  component,
  type,
  classes,
  gutterBottom,
  gutterTop,
  className,
  children
}) => {
  const Component = component || componentFromType(type);

  return (
    <Component className={classnames(
      className,
      classes[type],
      { [classes.gutterBottom]: gutterBottom },
      { [classes.gutterTop]: gutterTop }
    )}>
      {children}
    </Component>
  );
};

const styles = theme => {
  return {
    body1: {
      margin: 0
    },
    title: {
      margin: 0
    },
    gutterBottom: {
      marginBotom: theme.spacing.unit
    },
    gutterTop: {
      marginTop: theme.spacing.unit
    }
  };
};

const StyledText = injectStyle(styles)(Text);
StyledText.displayName = 'StyledText';
export default StyledText;

const componentFromType = key => ({
  body1: 'p',
  title: 'h1'
}[key] || 'span');
