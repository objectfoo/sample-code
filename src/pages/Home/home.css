.page-list {
  padding-left: 0;
  list-style-type: none;
}

.page-list a {
  display: block;
}
