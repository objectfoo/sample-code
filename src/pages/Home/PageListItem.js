import React from 'react';
import {Link} from 'react-router-dom';
import classnames from 'classnames';
import Text from '../../common/Text';

export const PageListItem = ({active, page, classes}) => {
  const liClassName = classnames(
    classes.pageListItem,
    { [classes['pageListItemActive']]: active }
  );

  return (
    <li className={liClassName}>
      <Text className={classes.pageLinkExample} type='body1' component='div'>
        {`Example: ${page.example}`}
      </Text>
      {active ? (
        <div className={classes.pageLinkTitle}>{page.title}</div>
      ) : (
        <Link
          to={page.path}
          className={classes.pageLinkTitle}>{page.title}</Link>
      )}
    </li>
  );
};

PageListItem.displayName = 'PageListItem';

export default PageListItem;
