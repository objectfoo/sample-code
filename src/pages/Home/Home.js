import React from 'react';
import injectStyle from 'react-jss';
import enhanceWithRoot from '../../common/enhanceWithRoot';
import PageListItem from './PageListItem';
import Text from '../../common/Text';

export const Home = ({pages = [], currentKey, classes, active}) => {
  return (
    <main className={classes.main}>
      <Text type='title'>Code samples</Text>
      <ul className={classes.pageList}>
        {pages.map(page => {
          return <PageListItem {...{
            key: page.key,
            active: page.key === currentKey,
            page,
            classes
          }} />;
        })}
      </ul>
    </main>
  );
};

const styles = ({spacing, typeography}) => {
  return {
    main: {
      padding: spacing.unit
    },
    pageList: {
      listStyleType: 'none',
      paddingLeft: 0
    },
    pageListItem: {
      display: 'flex'
    },
    pageListItemActive: {
      fontWeight: typeography.fontWeightMedium
    },
    pageLinkExample: {
      padding: spacing.unit / 2,
      flex: 'none'
    },
    pageLinkTitle: {
      padding: spacing.unit / 2,
      flex: 'auto'
    }
  };
};

export default enhanceWithRoot(
  injectStyle(styles)(Home)
);
