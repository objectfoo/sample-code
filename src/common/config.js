import Home from '../pages/Home/Home';
export {Home};

export const pageMap = {
  byKey: {
    home: {
      key: 'home',
      title: 'Page Index',
      component: Home,
      path: '/',
      exact: true,
      example: '1'
    },
    'code-sample-example': {
      key: 'code-sample-example',
      title: 'Code sample example',
      component: Home,
      path: '/code-sample-example',
      exact: true,
      example: '2'
    }
  },
  ordered: [
    'home',
    'code-sample-example'
  ]
};
