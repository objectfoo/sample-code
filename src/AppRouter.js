import React from 'react';
import { BrowserRouter as Router, Route } from 'react-router-dom';
import {pageMap} from './pages/config';

export const App = ({classes}) => {
  const orderedPages = getOrderedPages(pageMap);

  return (
    <Router>
      <div>
        {orderedPages.map(({key, path, exact, ...rest}) => {
          const routeP = { key, path, exact };
          return (
            <Route {...routeP} render={() => (
              <rest.component pages={orderedPages} currentKey={key} />
            )} />
          );
        })}
      </div>
    </Router>
  );
};

export default App;

export const getOrderedPages = pages =>
  pages.ordered.map(key => pages.byKey[key]);
