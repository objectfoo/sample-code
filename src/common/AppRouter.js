import React from 'react';
import {BrowserRouter as Router, Route} from 'react-router-dom';
import {pageMap} from './config';

export const AppRouter = () => {
  const orderedPages = getOrderedPages(pageMap);

  return (
    <Router>
      <div>
        {orderedPages.map(({key, path, exact, ...rest}) => {
          const routeP = { key, path, exact };
          return (
            <Route {...routeP} render={() => (
              <rest.component pages={orderedPages} currentKey={key} />
            )} />
          );
        })}
      </div>
    </Router>
  );
};

export default AppRouter;

export const getOrderedPages = pages =>
  pages.ordered.map(key => pages.byKey[key]);
