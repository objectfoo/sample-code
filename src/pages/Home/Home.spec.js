import React from 'react';
import {shallow} from 'enzyme';
import {Home} from './Home';
import {PageListItem} from './PageListItem';
import {getOrderedPages} from '../../common/AppRouter';
import {pageMap as mockPageMap} from '../../test/fixtures/mockPageMap';

describe('Text', () => {
  const mockClasses = {
    main: 'main',
    pageList: 'pageList',
    pageListItem: 'pageListItem',
    pageListItemActive: 'pageListItemActive',
    pageLinkExample: 'pageLinkExample',
    pageLinkTitle: 'pageLinkTitle'
  };

  it('should render a list of pages with a fancy title', () => {
    const wrapper = shallow(
      <Home
        classes={mockClasses}
        pages={getOrderedPages(mockPageMap)}
      />
    );
    expect(wrapper).toMatchSnapshot();
  });

  it('should render an active Page List Item', () => {
    const wrapper = shallow(
      <PageListItem
        active
        page={getOrderedPages(mockPageMap)[0]}
        classes={mockClasses} />
    );

    expect(wrapper).toMatchSnapshot();
  });

  it('should render an active Page List Item', () => {
    const wrapper = shallow(
      <PageListItem
        page={getOrderedPages(mockPageMap)[1]}
        classes={mockClasses} />
    );

    expect(wrapper).toMatchSnapshot();
  });
});
