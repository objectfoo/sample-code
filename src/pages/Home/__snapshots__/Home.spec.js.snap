// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Text should render a list of pages with a fancy title 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Unknown
    classes={
        Object {
            "main": "main",
            "pageLinkExample": "pageLinkExample",
            "pageLinkTitle": "pageLinkTitle",
            "pageList": "pageList",
            "pageListItem": "pageListItem",
            "pageListItemActive": "pageListItemActive",
          }
    }
    pages={
        Array [
            Object {
              "component": [Function],
              "exact": true,
              "example": "1",
              "key": "home",
              "path": "/",
              "title": "Page Index",
            },
            Object {
              "component": [Function],
              "exact": true,
              "example": "2",
              "key": "other-home",
              "path": "/other-home",
              "title": "Other Index",
            },
          ]
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <StyledText
          type="title"
>
          Code samples
</StyledText>,
        <ul
          className="pageList"
>
          <PageListItem
                    active={false}
                    classes={
                              Object {
                                        "main": "main",
                                        "pageLinkExample": "pageLinkExample",
                                        "pageLinkTitle": "pageLinkTitle",
                                        "pageList": "pageList",
                                        "pageListItem": "pageListItem",
                                        "pageListItemActive": "pageListItemActive",
                                      }
                    }
                    page={
                              Object {
                                        "component": [Function],
                                        "exact": true,
                                        "example": "1",
                                        "key": "home",
                                        "path": "/",
                                        "title": "Page Index",
                                      }
                    }
          />
          <PageListItem
                    active={false}
                    classes={
                              Object {
                                        "main": "main",
                                        "pageLinkExample": "pageLinkExample",
                                        "pageLinkTitle": "pageLinkTitle",
                                        "pageList": "pageList",
                                        "pageListItem": "pageListItem",
                                        "pageListItemActive": "pageListItemActive",
                                      }
                    }
                    page={
                              Object {
                                        "component": [Function],
                                        "exact": true,
                                        "example": "2",
                                        "key": "other-home",
                                        "path": "/other-home",
                                        "title": "Other Index",
                                      }
                    }
          />
</ul>,
      ],
      "className": "main",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": "Code samples",
          "type": "title",
        },
        "ref": null,
        "rendered": "Code samples",
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <PageListItem
              active={false}
              classes={
                            Object {
                                          "main": "main",
                                          "pageLinkExample": "pageLinkExample",
                                          "pageLinkTitle": "pageLinkTitle",
                                          "pageList": "pageList",
                                          "pageListItem": "pageListItem",
                                          "pageListItemActive": "pageListItemActive",
                                        }
              }
              page={
                            Object {
                                          "component": [Function],
                                          "exact": true,
                                          "example": "1",
                                          "key": "home",
                                          "path": "/",
                                          "title": "Page Index",
                                        }
              }
/>,
            <PageListItem
              active={false}
              classes={
                            Object {
                                          "main": "main",
                                          "pageLinkExample": "pageLinkExample",
                                          "pageLinkTitle": "pageLinkTitle",
                                          "pageList": "pageList",
                                          "pageListItem": "pageListItem",
                                          "pageListItemActive": "pageListItemActive",
                                        }
              }
              page={
                            Object {
                                          "component": [Function],
                                          "exact": true,
                                          "example": "2",
                                          "key": "other-home",
                                          "path": "/other-home",
                                          "title": "Other Index",
                                        }
              }
/>,
          ],
          "className": "pageList",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": "home",
            "nodeType": "function",
            "props": Object {
              "active": false,
              "classes": Object {
                "main": "main",
                "pageLinkExample": "pageLinkExample",
                "pageLinkTitle": "pageLinkTitle",
                "pageList": "pageList",
                "pageListItem": "pageListItem",
                "pageListItemActive": "pageListItemActive",
              },
              "page": Object {
                "component": [Function],
                "exact": true,
                "example": "1",
                "key": "home",
                "path": "/",
                "title": "Page Index",
              },
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": "other-home",
            "nodeType": "function",
            "props": Object {
              "active": false,
              "classes": Object {
                "main": "main",
                "pageLinkExample": "pageLinkExample",
                "pageLinkTitle": "pageLinkTitle",
                "pageList": "pageList",
                "pageListItem": "pageListItem",
                "pageListItemActive": "pageListItemActive",
              },
              "page": Object {
                "component": [Function],
                "exact": true,
                "example": "2",
                "key": "other-home",
                "path": "/other-home",
                "title": "Other Index",
              },
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
        ],
        "type": "ul",
      },
    ],
    "type": "main",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <StyledText
            type="title"
>
            Code samples
</StyledText>,
          <ul
            className="pageList"
>
            <PageListItem
                        active={false}
                        classes={
                                    Object {
                                                "main": "main",
                                                "pageLinkExample": "pageLinkExample",
                                                "pageLinkTitle": "pageLinkTitle",
                                                "pageList": "pageList",
                                                "pageListItem": "pageListItem",
                                                "pageListItemActive": "pageListItemActive",
                                              }
                        }
                        page={
                                    Object {
                                                "component": [Function],
                                                "exact": true,
                                                "example": "1",
                                                "key": "home",
                                                "path": "/",
                                                "title": "Page Index",
                                              }
                        }
            />
            <PageListItem
                        active={false}
                        classes={
                                    Object {
                                                "main": "main",
                                                "pageLinkExample": "pageLinkExample",
                                                "pageLinkTitle": "pageLinkTitle",
                                                "pageList": "pageList",
                                                "pageListItem": "pageListItem",
                                                "pageListItemActive": "pageListItemActive",
                                              }
                        }
                        page={
                                    Object {
                                                "component": [Function],
                                                "exact": true,
                                                "example": "2",
                                                "key": "other-home",
                                                "path": "/other-home",
                                                "title": "Other Index",
                                              }
                        }
            />
</ul>,
        ],
        "className": "main",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": "Code samples",
            "type": "title",
          },
          "ref": null,
          "rendered": "Code samples",
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <PageListItem
                active={false}
                classes={
                                Object {
                                                "main": "main",
                                                "pageLinkExample": "pageLinkExample",
                                                "pageLinkTitle": "pageLinkTitle",
                                                "pageList": "pageList",
                                                "pageListItem": "pageListItem",
                                                "pageListItemActive": "pageListItemActive",
                                              }
                }
                page={
                                Object {
                                                "component": [Function],
                                                "exact": true,
                                                "example": "1",
                                                "key": "home",
                                                "path": "/",
                                                "title": "Page Index",
                                              }
                }
/>,
              <PageListItem
                active={false}
                classes={
                                Object {
                                                "main": "main",
                                                "pageLinkExample": "pageLinkExample",
                                                "pageLinkTitle": "pageLinkTitle",
                                                "pageList": "pageList",
                                                "pageListItem": "pageListItem",
                                                "pageListItemActive": "pageListItemActive",
                                              }
                }
                page={
                                Object {
                                                "component": [Function],
                                                "exact": true,
                                                "example": "2",
                                                "key": "other-home",
                                                "path": "/other-home",
                                                "title": "Other Index",
                                              }
                }
/>,
            ],
            "className": "pageList",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": "home",
              "nodeType": "function",
              "props": Object {
                "active": false,
                "classes": Object {
                  "main": "main",
                  "pageLinkExample": "pageLinkExample",
                  "pageLinkTitle": "pageLinkTitle",
                  "pageList": "pageList",
                  "pageListItem": "pageListItem",
                  "pageListItemActive": "pageListItemActive",
                },
                "page": Object {
                  "component": [Function],
                  "exact": true,
                  "example": "1",
                  "key": "home",
                  "path": "/",
                  "title": "Page Index",
                },
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": "other-home",
              "nodeType": "function",
              "props": Object {
                "active": false,
                "classes": Object {
                  "main": "main",
                  "pageLinkExample": "pageLinkExample",
                  "pageLinkTitle": "pageLinkTitle",
                  "pageList": "pageList",
                  "pageListItem": "pageListItem",
                  "pageListItemActive": "pageListItemActive",
                },
                "page": Object {
                  "component": [Function],
                  "exact": true,
                  "example": "2",
                  "key": "other-home",
                  "path": "/other-home",
                  "title": "Other Index",
                },
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
          ],
          "type": "ul",
        },
      ],
      "type": "main",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`Text should render an active Page List Item 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <PageListItem
    active={true}
    classes={
        Object {
            "main": "main",
            "pageLinkExample": "pageLinkExample",
            "pageLinkTitle": "pageLinkTitle",
            "pageList": "pageList",
            "pageListItem": "pageListItem",
            "pageListItemActive": "pageListItemActive",
          }
    }
    page={
        Object {
            "component": [Function],
            "exact": true,
            "example": "1",
            "key": "home",
            "path": "/",
            "title": "Page Index",
          }
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <StyledText
          className="pageLinkExample"
          component="div"
          type="body1"
>
          Example: 1
</StyledText>,
        <div
          className="pageLinkTitle"
>
          Page Index
</div>,
      ],
      "className": "pageListItem pageListItemActive",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": "Example: 1",
          "className": "pageLinkExample",
          "component": "div",
          "type": "body1",
        },
        "ref": null,
        "rendered": "Example: 1",
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Page Index",
          "className": "pageLinkTitle",
        },
        "ref": null,
        "rendered": "Page Index",
        "type": "div",
      },
    ],
    "type": "li",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <StyledText
            className="pageLinkExample"
            component="div"
            type="body1"
>
            Example: 1
</StyledText>,
          <div
            className="pageLinkTitle"
>
            Page Index
</div>,
        ],
        "className": "pageListItem pageListItemActive",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": "Example: 1",
            "className": "pageLinkExample",
            "component": "div",
            "type": "body1",
          },
          "ref": null,
          "rendered": "Example: 1",
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Page Index",
            "className": "pageLinkTitle",
          },
          "ref": null,
          "rendered": "Page Index",
          "type": "div",
        },
      ],
      "type": "li",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`Text should render an active Page List Item 2`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <PageListItem
    classes={
        Object {
            "main": "main",
            "pageLinkExample": "pageLinkExample",
            "pageLinkTitle": "pageLinkTitle",
            "pageList": "pageList",
            "pageListItem": "pageListItem",
            "pageListItemActive": "pageListItemActive",
          }
    }
    page={
        Object {
            "component": [Function],
            "exact": true,
            "example": "2",
            "key": "other-home",
            "path": "/other-home",
            "title": "Other Index",
          }
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <StyledText
          className="pageLinkExample"
          component="div"
          type="body1"
>
          Example: 2
</StyledText>,
        <Link
          className="pageLinkTitle"
          replace={false}
          to="/other-home"
>
          Other Index
</Link>,
      ],
      "className": "pageListItem",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": "Example: 2",
          "className": "pageLinkExample",
          "component": "div",
          "type": "body1",
        },
        "ref": null,
        "rendered": "Example: 2",
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": "Other Index",
          "className": "pageLinkTitle",
          "replace": false,
          "to": "/other-home",
        },
        "ref": null,
        "rendered": "Other Index",
        "type": [Function],
      },
    ],
    "type": "li",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <StyledText
            className="pageLinkExample"
            component="div"
            type="body1"
>
            Example: 2
</StyledText>,
          <Link
            className="pageLinkTitle"
            replace={false}
            to="/other-home"
>
            Other Index
</Link>,
        ],
        "className": "pageListItem",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": "Example: 2",
            "className": "pageLinkExample",
            "component": "div",
            "type": "body1",
          },
          "ref": null,
          "rendered": "Example: 2",
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": "Other Index",
            "className": "pageLinkTitle",
            "replace": false,
            "to": "/other-home",
          },
          "ref": null,
          "rendered": "Other Index",
          "type": [Function],
        },
      ],
      "type": "li",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
